### DISCUSSION TITLE
Functional tests with Renovate container

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#23377] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Ability to execute functional tests on a running test environment generated from in-development code.

The test environment should be running renovate container. All third party services are virtualized/mocked.

Few questions:-
- The code for the image generation is tracked in a different repo.  Repeating the code in renovatebot/renovate repo is not a good idea.  The code won't be DRY.
Does it make sense to fold the docker-renovate code into renovate?

- Which framework to use for spinning the environment? There are multiple testing frameworks. Which one is a good fit?

- There are dedicated tools for service virtualization.  Does it make sense to pick one and use it?

### What you tried so far.

Nothing yet

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
- related: #23433
